### PR TITLE
use generated clients in test

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
We used hand rolled dialogue clients in our tests

## After this PR
Use conjure defined and generated clients

## Possible downsides?
It makes breaks to the generator target harder

### CODE REVIEW

File: dialogue-example/dialogue-example-dialogue/build/reports/checkstyle/main.html [hunk 1]
```diff
@@ -0,0 +1,119 @@
+<html>
```
Comment (carterkozak) [inline dialogue-example/dialogue-example-dialogue/build/reports/checkstyle/main.html]: I think we're missing `build/` in the .gitignore

Comment (ferozco) [reply dialogue-example/dialogue-example-dialogue/build/reports/checkstyle/main.html]: 🤦‍♂

**Comment by ferozco** [comment]
We'll probably want to wait on #475 merging and conjure-java adopting the change before merging this in

File: .gitignore [hunk 2]
```diff
@@ -8,6 +8,7 @@
 .node
 /build/
 */build/
+*/*/build
 examples/*/build/
 extras/*/build/
```
Comment (carterkozak) [inline .gitignore]: I think these two can be deleted since we have `*/*/build`

**Comment by ferozco** [comment]
Looks like my modules cache somehow got corrupted and was causing --write-locks to emit the wrong result 😨 Luckily `rm -rf ~/.gradle/caches/modules-2/` seems to have fixed it up.

**Comment by carterkozak** [comment]
Oh wow, next time I wonder if it's worth backing up the directory to root cause the corruption when we have cycles to dig into it

**Comment by carterkozak** [comment]
Thanks! :+1:

**Pull request merged**